Framework/API/src/Algorithm.cpp

```diff
@@ -32,6 +32,9 @@
 
 #include <map>
 
+// Index property handling template definitions
+#include "MantidAPI/Algorithm.tcc"
```

Comment:
Giving this another review pass for all the changes while I was on vacation. You did not manage to resolve this issue as suggested?